paddle/framework/op_registry.h

```diff
@@ -171,11 +179,27 @@ Add a mark to which output is temporary is helpful for future optimization.
       ++cnt;
     }
     PADDLE_ENFORCE(names.size() == cnt,
-                   "Cannot register two attribute in same name!");
+                   "Cannot register two attribute with same name!");
+  }
+
+  void CheckNoDuplicatedInOut() {
+    std::unordered_set<std::string> names;
+    size_t cnt = 0;
+    for (auto& input : proto_->inputs()) {
+      names.insert(input.name());
```

Comment:
Why not use:

```cpp
for (auto& input : proto_->inputs()) {
  std::string input_name = input.name();
  PADDLE_ENFORCE(!names.count(input_name), "%s is duplicated!", input_name);
  names.insert(input_name);
}
```

In this way, we are able to be informed about which one is duplicated.